<?xml version="1.0"?>
<launch>

  <arg name="use_rqt" default="true" />

  <group ns="camera">

  <node name="rqt_generic" pkg="rqt_gui" type="rqt_gui"
      args="--perspective-file $(find vimjay)/config/image_rect.perspective"
      if="$(arg use_rqt)"
      output="screen" >
  </node>

  <group if="true">
  <node name="dr_camera_info" pkg="vimjay" type="dr_camera_info.py"
      output="screen" >
    <remap from="camera_info" to="camera_info_in" />
  </node>

  <node name="image" pkg="vimjay" type="image.py"
      args="$(find vimjay)/data/chess.png"
      output="screen" >
    <param name="rate" value="5.0" />
  </node>
  </group>

  <group if="false">
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node"
      output="screen" >
    <param name="image_width" value="320" />
    <param name="image_height" value="240" />
    <param name="framerate" value="5" />
    <param name="video_device" value="/dev/video0" />
    <param name="camera_info_url" type="string" value="file://$(find vimjay)/config/camera.yaml" />
  </node>
  </group>

  <node name="image_proc" pkg="image_proc" type="image_proc"
      output="screen" >
    <remap from="image_raw" to="image" />
    <remap from="camera_info" to="camera_info_in" />
  </node>

  </group>  <!-- camera -->

</launch>
